Update the grid, make the orientation switch pretty and add project summary

randomer opened the pull request.

Commit:
Fix the summary and repository link

Commit:
Add the .git extension

Commit:
Human-understandable orientation switch

Commit:
Eye-pleasing grid

Matrix.Extra.prettyPrint doesn’t even compare…

Commit:
Visual enhancements

Commit:
Revert "Visual enhancements"

This reverts commit 7245c549692f660536b081860ddf882642547eb2.

Commit:
Revert "Revert "Visual enhancements""

This reverts commit 32815fb05504ab5057cc79d3bb1eac14e406efee.

john-kelly commented on Grid.elm:
```diff
@@ -10,22 +11,15 @@ import Matrix.Extra
 type Cell
     = Ship IsHit
     | Empty IsHit
+    -- TODO: Sunk state
```
What do you mean by this? The sunk state is handle by the `Ship IsHit`

randomer replied:
Not quite. There should be also an indication that a ship has been fully destroyed (every cell of a ship was hit)

john-kelly replied:
I see now. This makes sense.

randomer replied:
See? That's a good example why you should focus on the UI first — your imagination won't be enough to think of all the possibilities. Also, while we're on it, there doesn't need to be four fields in the Play view — two will suffice.

john-kelly commented on Grid.elm:
```diff
@@ -52,10 +46,52 @@ addShip ship grid =
     |> List.foldr setToShip grid
 
+cellToHtml : Cell -> Html.Html
+cellToHtml cell =
```
So, I'm all for making things look prettier, but not yet. The main focus right now is getting the functionality working. That's a big reason why I am using the `Matrix.prettyPrint` function in the first place. It allows a really nice temp ui while we are creating the game.

john-kelly commented on Main.elm:
```diff
@@ -73,30 +75,69 @@ setupControlsView address player =
   else Html.div [] []
 
+orientationSwitch : Signal.Address Action -> Int -> Ship.Orientation -> Html.Html
+orientationSwitch address shipId orientation =
```
So same here. I like the new look, but I think that it will probably get scrapped once we actually start focusing in on the ui. Right now the focus is getting the functionality working. Then we can take a step back and think about how we want to structure the ui -- both from a code standpoint and a interaction standpoint.

randomer replied:
Why would it? The thing is that functionality depends on the interface — depending on how you structure your UI, you will need different functions in order to change your model in different ways. That's why I think we should start working on the UI right away — top to bottom approach is more viable I think.

john-kelly replied:
What I mean is the code for "Switching a ships orientation" is all going to be the same, whether the ui is a radio button, two buttons, or a right mouse click event.

john-kelly replied:
The ui is more of a hook into the functionality. The functionality for changing a ships orientation (and many other functions) is independent from how the ui is implemented. For example, I see the ability to "CLICK" to add ships into the grid as a nice ui feature. That ui feature is going to use the same code for adding ships that the current ui uses.

randomer replied:
Not necessarily. For example, you're using four fields to display the game process. However, it's not needed — you need only two. But if you don't think about that, it eventually will affect your model, in which you'll have to take into account that you're using four fields.

john-kelly replied:
I'm not sure what you are referring to for this.

randomer replied:
This: https://github.com/john-kelly/elm-battleship/blob/master/Player.elm#L104
There needs to be only one grid per player

john-kelly replied:
So why is that?

randomer replied:
Long story short: http://kortaggio.github.io/battleboat/

john-kelly replied:
Okay. So I think that I have though about this one. The design reason for doing it this way is in preparation for networked gameplay. 

The problem with each player only having a single board is that they then need to know the state of their opponents full board in order to draw their tracking board.

This potentially presents a problem with cheating.

john-kelly replied:
The opponent could send their board to the server and then the server sends the tracking board to the client, removing the information pertaining to the opponents ship positions. But this is not related to ui at all. This is more of a design decision.

john-kelly commented:
@randomer Overall. I really like what you did here with the look. So beautiful. I just think that for right now we should get the core functionality working with the computer ai before thinking about ui.

As a side note, I think we should have a brainstorm about where css styles should live. I'm not sure the best way to keep styles maintainable in an elm project.

I am going to leave this PR open so we can reference it in the future.

randomer commented:
Thanks. I agree on the CSS point. There seems to be no way of including CSS when using elm-reactor

Commit:
People-oriented ship-placing

Commit:
Merge branch 'master' into ship-placement

Commit:
Merge branch 'ship-placement'

Commit:
Full-fledged StartApp

Effects will be needed in order to do networking and outside signals
(like key presses)

john-kelly committed:
refactor show and hide logic

Commit:
Merge branch 'john-kelly/refactor-show-hide-logic'

Commit:
Revert "Merge branch 'john-kelly/refactor-show-hide-logic'"

This reverts commit 7358e4a398c45f7e42ebac97f3f443fb82f5b1d0, reversing
changes made to 0b38052a4297e6c663c9e57ff64675946e230788.

Commit:
Merge branch 'ship-selector'

john-kelly commented:
@randomer Is this the most recent code?

randomer commented:
Yep. The latest version.

Commit:
Randomly-generated computer player

Commit:
Merge branch 'random-computer-player'

Commit:
Simplify the Fleet module

Commit:
We can shoot stuff now!

Commit:
Merge branch 'simple-fleet' into shootable-cells

Commit:
Merge branch 'shootable-cells'

john-kelly committed:
some styling, refactor hide and show ship

Commit:
Sinkable ships

Commit:
Merge remote-tracking branch 'john-kelly/refactor-hide-show-ship' into shootable-cells

Grid show/hide logic simplification

Commit:
Merge branch 'shootable-cells'
